fs-tep-api/src/main/java/com/cgi/eoss/fstep/api/controllers/UserPreferencesApiImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.cgi.eoss.fstep.api.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Component;
+
+import com.cgi.eoss.fstep.model.QUser;
+import com.cgi.eoss.fstep.model.QUserPreference;
+import com.cgi.eoss.fstep.model.UserPreference;
+import com.cgi.eoss.fstep.persistence.dao.UserPreferenceDao;
+import com.cgi.eoss.fstep.security.FstepSecurityService;
+import com.querydsl.core.types.dsl.NumberPath;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Getter
+@Component
+public class UserPreferencesApiImpl extends BaseRepositoryApiImpl<UserPreference> implements UserPreferencesApiCustom  {
+
+    private static final int MAX_PREFERENCES_PER_USER = 200;
+	private final FstepSecurityService securityService;
+    private final UserPreferenceDao dao;
+    
+    
+    @Override
+    public <S extends UserPreference> S save(S entity) {
+        if (entity.getOwner() == null) {
+            getSecurityService().updateOwnerWithCurrentUser(entity);
+        }
+        //Allow max items per user
+        int currentPreferenceCount = getDao().findByOwner(entity.getOwner()).size();
+        if (currentPreferenceCount >= MAX_PREFERENCES_PER_USER) {
+        		throw new org.springframework.security.access.AccessDeniedException("403 returned");
```

Comment:
I think the AccessDeniedException will lead to 403 already - the message here could be more meaningful.